llvm/lib/CAS/CASDB.cpp

```diff
@@ -138,3 +138,21 @@ CASDB::loadIndependentDataBuffer(ObjectHandle Node, const Twine &Name,
   return std::make_unique<SmallVectorMemoryBuffer>(std::move(Bytes), Name.str(),
                                                    NullTerminate);
 }
+
+
+Expected<std::unique_ptr<CASDB>> cas::createCASFromIdentifier(StringRef Path) {
+  if (Path.empty() || Path.startswith("mem://"))
+    return createInMemoryCAS();
+
+  if (Path.consume_front("plugin://"))
+    return createPluginCAS(Path);
+
+  if (Path.consume_back("file://"))
+    return createOnDiskCAS(Path);
```

Comment:
If this is an actual URL we probably need to handle percent encoding (e.g. %20 for space).  Same for `plugin://` above.  I'm fine with just adding a FIXME comment for now.

Reply:
Ok. If we stick with URL schema, we can pass plugin arguments as:
`plugin://address?arguments` but we need to support the URL correctly for that.